tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/call_settings.py

```diff
@@ -0,0 +1,211 @@
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._sim_pin_area_locator))
+
+    def tap_auth_numbers(self):
+        element = self.marionette.find_element(*self._auth_number_locator)
+        Wait(self.marionette).until(expected.element_displayed(element)and expected.element_enabled(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._auth_number_page_locator))
+
+    def tap_call_forwarding(self):
+        element = self.marionette.find_element(*self._call_forwarding_locator)
+        Wait(self.marionette).until(expected.element_displayed(element)and expected.element_enabled(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._cf_page_locator))
+
+    # the default is 'disabled'
+    def wait_until_cf_info_received(self):
+        Wait(self.marionette,timeout=20).until(expected.element_displayed(*self._cf_status_disabled_text_locator))
+
+    def tap_always_forward(self):
+        element = self.marionette.find_element(*self._call_forwarding_always_locator)
+        Wait(self.marionette).until(expected.element_displayed(element)and expected.element_enabled(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._cf_always_page_locator))
+
+    def tap_forward_when_busy(self):
+        element = self.marionette.find_element(*self._call_forwarding_busy_locator)
+        Wait(self.marionette).until(expected.element_displayed(element)and expected.element_enabled(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._cf_busy_page_locator))
+
+    def tap_forward_unanswered(self):
+        element = self.marionette.find_element(*self._call_forwarding_unanswered_locator)
+        Wait(self.marionette).until(expected.element_displayed(element)and expected.element_enabled(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._cf_unanswered_locator))
+
+    def tap_forward_unreachable(self):
+        element = self.marionette.find_element(*self._call_forwarding_unreachable_locator)
+        Wait(self.marionette).until(expected.element_displayed(element) and expected.element_enabled(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._cf_unreachable_locator))
+
+    def tap_call_barring(self):
+        element = self.marionette.find_element(*self._call_barring_locator)
+        Wait(self.marionette).until(expected.element_displayed(element)and expected.element_enabled(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._cb_page_locator))
+
+    def wait_until_cb_info_received(self):
+        status = self.marionette.find_element(*self._cb_status_text_locator)
+        Wait(self.marionette).until(lambda m: status.get_attribute('data-l10n-id') == 'disabled')
```

Comment:
Ok, you need to add a timeout=30 here, that helps at least for me to fix the failure I was describing extensively in the bug.